### PR TITLE
Add automatic rig ID detection

### PR DESCRIPTION
**Pull request by TheBlueBurger**

### CODE REVIEW

File: setup.js [hunk 1]
```diff
@@ -30,19 +31,59 @@ async function continueSetup(clear) {
...
     }]);
-    if (useapi.useapi == "api") {
+    if(promptResult.useapi == "auto") {
+        let logPath;
+        if(process.platform == "win32") {
+            logPath = path.join(process.env.APPDATA, "Salad", "logs", "main.log");
+        } else if(process.platform == "linux") {
+            logPath = path.join(process.env.HOME, ".config", "Salad", "logs", "main.log");
+        } else if(process.platform == "darwin") {
+            logPath = path.join(process.env.HOME, "Library", "Logs", "Salad", "main.log"); // untested, google says it works
```
Comment (vtheskeleton) [inline setup.js:58]: lmao

**Comment by TheBlueBurger** [comment]
I'll test this on my linux VM

File: setup.js [hunk 2]
```diff
@@ -30,19 +31,59 @@ async function continueSetup(clear) {
...
         message: "How would you like to provide your mining details?",
-        choices: [{
-            name: `Automatic ${chalk.yellow("(Salad auth token required!)")}`,
+        choices: [
+            {
+                name: `Automatic ${chalk.yellow("(Read from salad logs)")}`,
+                value: "auto"
+            },
+            {
+            name: `Fetch from account ${chalk.yellow("(Salad auth token required!)")}`,
```
Comment (vtheskeleton) [inline setup.js:44]: I would rather this isnt renamed

**Review by vtheskeleton — CHANGES_REQUESTED** [review]
Do not rename the original automatic option

**Comment by Tpelaaja** [comment]
Why remove the Auth Code method?

This is gonna bork if the log files dont have the current wallet address as they regenerate sometimes

Should be an third option

**Comment by vtheskeleton** [comment]
> Why remove the Auth Code method?
> 
> This is gonna bork if the log files dont have the current wallet address as they regenerate sometimes
> 
> Should be an third option

it isnt removed?

File: setup.js [hunk 3]
```diff
@@ -30,19 +31,59 @@ async function continueSetup(clear) {
...
+            logFileContent = fs.readFileSync(logPath).toString();
+        } catch(err) {
+            console.log(chalk.bold.red("A error occured while reading the log files, make sure that you have ran Salad and that SaladBind has permission to access it."))
+        }
+        const rigIDRegex = /^NiceHash rig ID: [a-z0-9]{15}$/m;
+        let rigID = logFileContent.match(rigIDRegex).join(" ");
+        if(rigID) rigID = rigID.split(": ")[1];
+        if(!rigID) {
+            console.log(chalk.bold.red("Could not find your Rig ID! Please make sure that you have mined for at least 5 minutes using Salad's official application."));
+            setTimeout(() => require("./index").menu(true), 3500)
```
Comment (vtheskeleton) [inline setup.js:71]: we shouldnt return to the menu here

**Review by vtheskeleton — CHANGES_REQUESTED** [review]

File: setup.js (hunk 3 above)
Comment (vtheskeleton) [reply setup.js:73]: then they will be in the menu with no config file

Comment (TheBlueBurger) [reply setup.js:73]: Ah ok ill fix that

**Review by vtheskeleton — APPROVED** [review]
Looks good, this will be included in the next patch

**Pull request merged**